sw/device/tests/sim_dv/sysrst_ctrl_ec_rst_l_test.c

```diff
@@ -0,0 +1,225 @@
+  // Issue WFI and wait for reset condition.
+  test_status_set(kTestStatusInWfi);
+  wait_for_interrupt();
+}
+
+// Waits for the kTestPhase variable to be changed by a backdoor overwrite
+// from the testbench in chip_sw_sysrst_ctrl_ec_rst_l_vseq.sv. This will
+// indicate that the testbench is ready to proceed with the
+// next phase of the test.
+static void wait_next_test_phase(void) {
+  uint8_t current_phase = kTestPhase;
+  // Set WFI status for testbench synchronization,
+  // no actual WFI instruction is issued.
+  test_status_set(kTestStatusInWfi);
+  test_status_set(kTestStatusInTest);
+  IBEX_SPIN_FOR(current_phase != kTestPhase, kTestPhaseTimeoutUsec);
+}
+
+// Enables the sysrst_ctrl overrides for the output pins. Allows
+// both low and high override values.
+static void override_setup(uint8_t pins_to_override) {
+  for (int i = 0; i < kOutputNumPads; ++i) {
+    if ((pins_to_override >> i) & 0x1) {
+      CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_set_enabled(
+          &sysrst_ctrl, kSysrstCtrlOutputs[i], kDifToggleEnabled));
+      CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_set_allowed(
+          &sysrst_ctrl, kSysrstCtrlOutputs[i], true, true));
+    }
+  }
+}
+
+// Disables the overrides. Allows the outputs to pass-through the
+// values from the relevant input pins.
+static void override_disable(void) {
+  for (int i = 0; i < kOutputNumPads; ++i) {
+    CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_set_enabled(
+        &sysrst_ctrl, kSysrstCtrlOutputs[i], kDifToggleDisabled));
+  }
+}
+
+// Sets the values of the output overrides as required.
+static void set_output_overrides(uint8_t override_value) {
+  for (int i = 0; i < kOutputNumPads; ++i) {
+    CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_set_override(
+        &sysrst_ctrl, kSysrstCtrlOutputs[i], (override_value >> i) & 0x1));
+  }
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_sysrst_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_SYSRST_CTRL_AON_BASE_ADDR),
+      &sysrst_ctrl));
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  pinmux_setup();
+  rstmgr_reset_info = rstmgr_testutils_reason_get();
+  CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_set_enabled(
+      &sysrst_ctrl, kDifSysrstCtrlPinEcResetInOut, kDifToggleDisabled));
+
+  while (kTestPhase < kTestPhaseDone) {
+    switch (kTestPhase) {
+      case kTestPhaseSetup:
+        CHECK(rstmgr_reset_info == kDifRstmgrResetInfoPor);
+        check_combo_reset();
+        return true;
+      case kTestPhaseCheckComboReset:
+        CHECK(rstmgr_reset_info == kDifRstmgrResetInfoSysRstCtrl);
+        break;
+      case kTestPhaseLoopback:
+        break;
```

Comment:
we don't have a loopback phase in this test. Can we remove it?

Reply:
Fixed